docs/proposal/multi-target.md

```diff
@@ -0,0 +1,119 @@
+then it is up to the provider to make sure that the change list of records passed to the DNS provider API is valid. **TODO**: explain best strategy.    
+
+Additionally see https://github.com/kubernetes-incubator/external-dns/issues/258
+
+## Implementation plan
+
+Brief summary of open PRs and what they are trying to address:
+
+### PRs 
+
+1. https://github.com/kubernetes-incubator/external-dns/pull/243 - first attempt to add support for multiple targets. It is lagging far behind from master tip
+    
+    *what it does*: unfinished attempt to extend `Endpoint` struct, for it to allow multiple targets (essentially `target string -> targets []string`)
+    
+    *action*: evaluate if rebasing makes sense, or we can just close it. 
+    
+2. https://github.com/kubernetes-incubator/external-dns/pull/261 - attempt to rework `plan` to make it work correctly with multiple targets. 
+    
+    *what it does* : attempts to fix issues with `plan` described in `Current Behaviour` section above. Included tests reveal the current problem with `plan`
+    
+    *action*: rebase on master and make necessary changes to satisfy requirements listed in this document 
+    
+3. https://github.com/kubernetes-incubator/external-dns/pull/326 - attempt to add multiple target support. 
+    
+    *what it does*: for each pair `DNS Name` + `Record Type` it aggregates **all** targets from the cluster and passes them to Provider. It adds basic support
+    for DO, Azura, Cloudflare, AWS, GCP, however those are not tested (?). (DNSSimple and Infoblox providers were not updated)
+    
+    *action*: the `plan` logic will probably needs to be reworked, however the rest concerning support in Providers and extending `Endpoint` struct can be reused. 
+    Rebase on master and add missing pieces. Depends on `2`. 
+    
+ Related PRs: https://github.com/kubernetes-incubator/external-dns/pull/331/files,  https://github.com/kubernetes-incubator/external-dns/pull/347/files - aiming at AWS Route53 weighted records.
+These PRs should be considered after common agreement about the way to address multi-target support is achieved. Related discussion:  https://github.com/kubernetes-incubator/external-dns/issues/196
+
+### How to proceed from here
+
+The following steps are needed: 
+1. Make sure consensus regarding the approach is achieved via collaboration on the current document 
+2. Notify all PR (see above) authors about the agreed approach
+3. Implementation: 
+    
+    a. `Plan` is working as expected - either based on #261 above or from scratch. `Plan` should be working correctly regardless of multi-target support
+    
+    b. Extensive testing making sure new `plan` does not introduce any breaking changes
+    
+    c. Change Endpoint struct to support multiple targets - based on #326 - integrate it with new `plan` @sethpollack
+    
+    d. Make sure new endpoint format can still be used in providers which have only partial support for multi targets **TODO**: how ? 
```

Comment:
Assuming the endpoint will have targets defined as `[]string` could we just do an initial implementation which uses `targets[0]` for providers that doesn't support multi targets?

In the long run you probably want to select the most correct target in the list *somehow* but given multi target support won't be supported fully anyway for those providers, just make it best effort in the first iteration IMO.

Reply:
agreed

Reply:
yeah let's go with this